test/benchmarks/mdc/checkbox/checkbox.perf-spec.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {$, browser} from 'protractor';
+import {runBenchmark} from '@angular/dev-infra-private/benchmark/driver-utilities';
+
+describe('checkbox performance benchmarks', () => {
+  beforeAll(() => {
+    browser.rootEl = '#root';
+  });
+
+  it('renders a checked checkbox', async() => {
+    await runBenchmark({
+      id: 'checkbox-render-checked',
+      url: '',
+      ignoreBrowserSynchronization: true,
+      params: [],
+      setup: async () => {
+        await $('#show').click();
+        await $('mat-checkbox').click();
+      },
+      prepare: async () => {
+        expect(await $('mat-checkbox input').isSelected())
+          .toBe(true, 'The checkbox should be in a selected state.');
+        await $('#hide').click();
+      },
+      work: async () => await $('#show').click()
+    });
+  });
+
+  it('renders an unchecked checkbox', async() => {
+    await runBenchmark({
+      id: 'checkbox-render-unchecked',
+      url: '',
+      ignoreBrowserSynchronization: true,
+      params: [],
+      setup: async() => await $('#show').click(),
+      prepare: async () => {
+        expect(await $('mat-checkbox input').isSelected())
+          .toBe(false, 'The checkbox should be in an unselected state.');
+        await $('#hide').click();
+      },
+      work: async () => await $('#show').click()
+    });
+  });
+
+  it('renders an indeterminate checkbox', async() => {
+    await runBenchmark({
+      id: 'checkbox-render-indeterminate',
+      url: '',
+      ignoreBrowserSynchronization: true,
+      params: [],
+      setup: async() => {
+        await $('#show').click();
+        await $('#indeterminate').click();
+      },
+      prepare: async () => {
+        expect(await $('mat-checkbox input').getAttribute('indeterminate'))
+          .toBe('true', 'The checkbox should be in an indeterminate state');
+        await $('#hide').click();
+      },
+      work: async () => await $('#show').click()
+    });
+  });
+
+  it('updates from unchecked to checked', async() => {
+    await runBenchmark({
+      id: 'checkbox-click-unchecked-to-checked',
+      url: '',
+      ignoreBrowserSynchronization: true,
+      params: [],
+      setup: async () => {
+        await $('#show').click();
+        await $('mat-checkbox').click();
+      },
+      prepare: async () => {
+        await $('mat-checkbox').click();
```

Comment:
Out of curiosity: Are those `clicks` unnecessary here? When the page loads, the checkbox should be already unchecked, right? Same would apply to other benchmarks here.

Reply:
The page is only loaded once for each benchmark. It goes 1. load page 2. call `setup` 3. call `prepare` 4. call `work` 5. repeat 3 and 4 over and over again.

Reply:
I see, how about having a reset button instead then? It feels not ideal to me that we need to check it initially, and then uncheck it again.

Reply:
I think adding a reset button would be a bit overkill here. I think just clicking once before the prepare/work cycle should be sufficient

Reply:
Either way works for me, but personally I'd not consider this as readable/clear. It all starts with the ambiguity of `prepare` and `setup`.